msm/src/circuit_design/capabilities.rs

```diff
@@ -11,17 +11,25 @@ use ark_ff::PrimeField;
 /// Environment capability for accessing and reading columns. This is necessary for
 /// building constraints.
 pub trait ColAccessCap<F: PrimeField, CIx: ColumnIndexer> {
+    // NB: 'static here means that `Variable` does not contain any
+    // references with a lifetime less than 'static. Which is true in
+    // our case. Necessary for `set_assert_mapper`
     type Variable: Clone
         + std::ops::Add<Self::Variable, Output = Self::Variable>
         + std::ops::Sub<Self::Variable, Output = Self::Variable>
         + std::ops::Mul<Self::Variable, Output = Self::Variable>
         + std::ops::Neg<Output = Self::Variable>
         + From<u64>
-        + std::fmt::Debug;
+        + std::fmt::Debug
+        + 'static;
```

Comment:
:thinking: not sure I like this, but let's see. What will have `'static` lifetime?
A value with `'static` lifetime will be embedded in the read-only data segment of the binary.